mozjs/src/jsimpls.rs

```diff
@@ -380,20 +381,117 @@ impl<T> JS::Rooted<T> {
 }
 
 impl JS::ObjectOpResult {
+    pub fn ok(&self) -> bool {
+        assert_ne!(self.code_, JS::ObjectOpResult_SpecialCodes::Uninitialized as usize);
+        self.code_ == JS::ObjectOpResult_SpecialCodes::OkCode as usize
+    }
+
```

Comment:
How is this different than `succeed` other than the assertion?

Reply:
`succeed` sets the code to `OkCode`. This just checks that its not unitialised and if its `OkCode`. It follows the same API as the C++ method.